Performing C SOURCE FILE Test CMAKE_HAVE_LIBC_PTHREAD failed with the following output:
Change Dir: /home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_2442b/fast && /usr/bin/make  -f CMakeFiles/cmTC_2442b.dir/build.make CMakeFiles/cmTC_2442b.dir/build
make[1]: 进入目录“/home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp”
Building C object CMakeFiles/cmTC_2442b.dir/src.c.o
/usr/bin/cc -DCMAKE_HAVE_LIBC_PTHREAD   -o CMakeFiles/cmTC_2442b.dir/src.c.o -c /home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_2442b
/home/ucar/.local/lib/python3.6/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_2442b.dir/link.txt --verbose=1
/usr/bin/cc -rdynamic CMakeFiles/cmTC_2442b.dir/src.c.o -o cmTC_2442b 
CMakeFiles/cmTC_2442b.dir/src.c.o：在函数‘main’中：
src.c:(.text+0x48)：对‘pthread_create’未定义的引用
src.c:(.text+0x50)：对‘pthread_detach’未定义的引用
src.c:(.text+0x58)：对‘pthread_cancel’未定义的引用
src.c:(.text+0x64)：对‘pthread_join’未定义的引用
src.c:(.text+0x74)：对‘pthread_atfork’未定义的引用
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_2442b.dir/build.make:98: recipe for target 'cmTC_2442b' failed
make[1]: *** [cmTC_2442b] Error 1
make[1]: 离开目录“/home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp”
Makefile:127: recipe for target 'cmTC_2442b/fast' failed
make: *** [cmTC_2442b/fast] Error 2


Source file was:
#include <pthread.h>

static void* test_func(void* data)
{
  return data;
}

int main(void)
{
  pthread_t thread;
  pthread_create(&thread, NULL, test_func, NULL);
  pthread_detach(thread);
  pthread_cancel(thread);
  pthread_join(thread, NULL);
  pthread_atfork(NULL, NULL, NULL);
  pthread_exit(NULL);

  return 0;
}

Determining if the function pthread_create exists in the pthreads failed with the following output:
Change Dir: /home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/make -f Makefile cmTC_f1abc/fast && /usr/bin/make  -f CMakeFiles/cmTC_f1abc.dir/build.make CMakeFiles/cmTC_f1abc.dir/build
make[1]: 进入目录“/home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp”
make[1]: 警告：文件“/home/ucar/.local/lib/python3.6/site-packages/cmake/data/share/cmake-3.23/Modules/CheckFunctionExists.c”的修改时间在未来 20510097 秒后
Building C object CMakeFiles/cmTC_f1abc.dir/CheckFunctionExists.c.o
/usr/bin/cc   -DCHECK_FUNCTION_EXISTS=pthread_create -o CMakeFiles/cmTC_f1abc.dir/CheckFunctionExists.c.o -c /home/ucar/.local/lib/python3.6/site-packages/cmake/data/share/cmake-3.23/Modules/CheckFunctionExists.c
Linking C executable cmTC_f1abc
/home/ucar/.local/lib/python3.6/site-packages/cmake/data/bin/cmake -E cmake_link_script CMakeFiles/cmTC_f1abc.dir/link.txt --verbose=1
/usr/bin/cc  -DCHECK_FUNCTION_EXISTS=pthread_create -rdynamic CMakeFiles/cmTC_f1abc.dir/CheckFunctionExists.c.o -o cmTC_f1abc  -lpthreads 
/usr/bin/ld: 找不到 -lpthreads
collect2: error: ld returned 1 exit status
CMakeFiles/cmTC_f1abc.dir/build.make:98: recipe for target 'cmTC_f1abc' failed
make[1]: *** [cmTC_f1abc] Error 1
make[1]: 离开目录“/home/ucar/catkin_ws/build/CMakeFiles/CMakeTmp”
Makefile:127: recipe for target 'cmTC_f1abc/fast' failed
make: *** [cmTC_f1abc/fast] Error 2



